core/amber/src/main/scala/edu/uci/ics/texera/web/resource/WorkflowWebsocketResource.scala

```diff
@@ -248,4 +241,23 @@ class WorkflowWebsocketResource {
 
   }
 
+  @OnClose
+  def myOnClose(session: Session, cr: CloseReason): Unit = {
+    if (WorkflowWebsocketResource.sessionJobs.contains(session.getId)) {
+      println(s"session ${session.getId} disconnected, kill its controller actor")
+      this.killWorkflow(session)
+    }
+
+    sessionResults.remove(session.getId)
```

Comment:
do we need to clear config object on close?

Reply:
it will be removed along with websocketSession now. (by GC in sessionMap)